lib/widgets/lightbox.dart

```diff
@@ -197,22 +199,207 @@ class _LightboxPageState extends State<_LightboxPage> {
+  final Uri src;
+
+  @override
+  State<VideoLightboxPage> createState() => _VideoLightboxPageState();
+}
+
+class _VideoLightboxPageState extends State<VideoLightboxPage> {
+  VideoPlayerController? _controller;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    assert(debugLog('_VideoLightboxPageState($hashCode).didChangeDependencies'));
+
+    // We are pretty sure that the dependencies (PerAccountStore & ZulipLocalizations)
+    // won't change while user is on the lightbox page but we still
+    // handle the reinitialization for correctness.
+    if (_controller != null) _deinitialize();
+    _initialize();
+  }
+
+  Future<void> _initialize() async {
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    assert(debugLog('VideoPlayerController.networkUrl(${widget.src})'));
+    _controller = VideoPlayerController.networkUrl(widget.src, httpHeaders: {
+      if (widget.src.origin == store.account.realmUrl.origin) ...authHeader(
+        email: store.account.email,
+        apiKey: store.account.apiKey,
+      ),
+      ...userAgentHeader()
+    });
+    _controller!.addListener(_handleVideoControllerUpdates);
+
+    try {
+      await _controller!.initialize();
+      await _controller!.play();
+    } catch (error) { // TODO(log)
+      assert(debugLog("VideoPlayerController.initialize failed: $error"));
+      if (mounted) {
+        await showErrorDialog(
+          context: context,
+          title: zulipLocalizations.errorDialogTitle,
+          message: zulipLocalizations.errorVideoPlayerFailed,
+          // To avoid showing the disabled video lightbox for the unnsupported
+          // video, we make sure user doesn't reach there by dismissing the dialog
+          // by clicking around it, user must press the 'OK' button, which will
+          // take user back to content message list.
+          barrierDismissible: false,
+          onContinue: () {
+            Navigator.pop(context); // Pops the dialog
+            Navigator.pop(context); // Pops the lightbox
+          });
+      }
+    }
+  }
+
+  @override
+  void dispose() {
+    _deinitialize();
+    super.dispose();
+  }
+
+  void _deinitialize() {
+    _controller?.removeListener(_handleVideoControllerUpdates);
+    _controller?.dispose();
+    _controller = null;
+  }
+
+  void _handleVideoControllerUpdates() {
+    setState(() {});
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _LightboxPageLayout(
+      routeEntranceAnimation: widget.routeEntranceAnimation,
+      message: widget.message,
+      buildBottomAppBar: (context, color, elevation) =>
+        _controller == null
+          ? null
+          : BottomAppBar(
+              height: 150,
+              color: color,
+              elevation: elevation,
+              child: Column(mainAxisAlignment: MainAxisAlignment.end, children: [
+              Row(children: [
```

Comment:
child should be indented:
```suggestion
              child: Column(mainAxisAlignment: MainAxisAlignment.end, children: [
                Row(children: [
```

(as should a list element in general)

I'd usually say "nit" for formatting issues; but this is a big widget expression with a lot of nesting and multiple lists, and so between this and a couple of other formatting issues it actually gets noticeably hard to read the structure of it.